.github/workflows/ci.yml

```diff
@@ -181,6 +195,43 @@ jobs:
         echo "############################################"
         printenv
 
+    # i386 build on x86_64 only requires passing -m32 to CFLAGS & LDFLAGS
+    - name: Meson init for i386
+      if: ${{ matrix.variant == 'i386' }}
+      run: |
+        mkdir build
+        CFLAGS="-m32" LDFLAGS="-m32" meson setup . build
+
+    - name: Meson init with cross compile
+      if: ${{ matrix.variant == 'cross-compile' }}
+      run: |
+        # Generate cross compile file (see https://mesonbuild.com/Cross-compilation.html#cross-compilation)
```

Comment:
surely this can be done with a single `echo` and some multiline magic?

Reply:
This is cleaner. than magic, so it is better. What can be done is `cat << EOF .... EOF`, but for me this code is fine too.

Reply:
I wasn’t sure it would work from a GH action yaml but I can try

Reply:
Well it doesn't work due to how the commands are passed from the yaml file